Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -263,24 +228,57 @@ extension TopSitesPanel: UICollectionViewDelegate {
     }
 
     func collectionView(collectionView: UICollectionView, willDisplayCell cell: UICollectionViewCell, forItemAtIndexPath indexPath: NSIndexPath) {
-        if let thumbnailCell = cell as? ThumbnailCell {
-            thumbnailCell.delegate = self
-
-            if editingThumbnails && indexPath.item < dataSource.data.count && thumbnailCell.removeButton.hidden {
-                thumbnailCell.removeButton.hidden = false
-            }
-        }
+        let thumbnailCell = cell as? ThumbnailCell
+        thumbnailCell?.delegate = self
+        updateRemoveButtonStateForIndexPath(indexPath, forCell: thumbnailCell)
     }
 }
 
 extension TopSitesPanel: ThumbnailCellDelegate {
+
     func didRemoveThumbnail(thumbnailCell: ThumbnailCell) {
-        if let indexPath = collection?.indexPathForCell(thumbnailCell) {
-            if let site = dataSource[indexPath.item] {
-                self.deleteHistoryTileForSite(site, atIndexPath: indexPath)
+        guard let indexPath = collection?.indexPathForCell(thumbnailCell),
+            let site = dataSource[indexPath.item] else {
+            return
+        }
+
+        if !refillingTiles {
```

Comment:
I'm trying to think of when this will be true. When they exit edit mode, immediately re-enter it and remove another tile?